src/components/TradeWidget/TokenRow.tsx

```diff
@@ -141,13 +143,9 @@ function displayBalance(balance: TokenBalanceDetails | undefined | null, key: st
   return formatAmount(balance[key], balance.decimals) || '0'
 }
 
-function getMax(balance: TokenBalanceDetails | null, token: TokenDetails): string {
-  return formatAmountFull((balance || {}).exchangeBalance, token.decimals, false) || '0'
-}
-
 // TODO: move into a validators file?
-function validatePositive(value: string): true | string {
-  return Number(value) > 0 || 'Invalid amount'
+function validatePositive(value: string, precision: number): true | string {
+  return new BN(parseAmount(value, precision) || '0').gt(ZERO) || 'Invalid amount'
```

Comment:
Do we really care about precision when comparing to 0? `1.123456789045678 > 0`  same as `1.1234 > 0`

Reply:
We do if using BN. In fact we also do if dealing with Number due to issues with float point.
Example for BN:

```typescript
parseAmount('0.001', 2) === new BN('0.00'); === ZERO
parseAmount('0.001', 3) === new BN('0.001'); > ZERO
```

Reply:
Well, actually, if we are enforcing the precision, the example where I do need precision will never happen. In fact, I can don't need BN at all here. As long as the value is a valid number > 0 it should return true.

I'll revert this part.